### PR TITLE
Add an option to log outcome of loading Ready to Run images

### PR DESCRIPTION
**Pull request by JohnChen0**

### CODE REVIEW

**Comment by JohnChen0** [comment]
@jkotas @fadimounir @gkhanna79 PTAL

**Comment by JohnChen0** [comment]
CI doesn't appear to be working properly.

**Comment by gkhanna79** [comment]
LGTM

**Comment by gkhanna79** [comment]
@mmitche CI is down?

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

File: src/vm/readytoruninfo.cpp [hunk 1]
```diff
@@ -351,6 +351,56 @@ BOOL ReadyToRunInfo::IsReadyToRunEnabled()
...
+            if (r2rLogFile != NULL)
+                fclose(r2rLogFile);
+            r2rLogFile = s_r2rLogFile;
+        }
+    }
+
+    if (r2rLogFile == NULL)
+        return;
+
+    fprintf(r2rLogFile, "%s: \"%s\".\n", msg, pFile->GetSimpleName());
```
Comment (jkotas) [inline src/vm/readytoruninfo.cpp]: It may be useful to log the full path here.

**Comment by jkotas** [comment]
@brianrob Is this something that would be useful to log in ETW traces?

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by fadimounir** [comment]
LGTM

**Comment by brianrob** [comment]
Yes, this would be very useful to have in ETW traces.  We use the fusion log on desktop to understand why NGEN images aren't loaded and it has proven to be quite useful.

**Comment by JohnChen0** [comment]
@dotnet-bot test Windows_NT Checked pri1r2r please

**Comment by JohnChen0** [comment]
@dotnet-bot test Ubuntu Checked pri1r2r please

**Comment by mmitche** [comment]
When you all get errors, please investigate and look at the logs to determine what the failure is.  If you don't know, then please post the link to the log/build.  Asking whether the CI is down is not useful;. @gkhanna79 What failed?

The OSX failure appears to be MyGet related.  https://twitter.com/MyGetTeam/status/740976282586996741

**Comment by gkhanna79** [comment]
@mmitche husdon issue, merge conflict, etc.

**Comment by mmitche** [comment]
Which one was a hudson issue?

**Pull request merged**